Detectors/TRD/base/src/LTUParam.cxx

```diff
@@ -0,0 +1,252 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+#include <stdio.h>
+#include "TMath.h"
+#include "TRDBase/LTUParam.h"
+#include "fairlogger/Logger.h"
+
+using namespace o2::trd;
+
+// definition of geometry constants
+std::array<float, 30> LTUParam::mgZrow = {
+  301, 177, 53, -57, -181,
+  301, 177, 53, -57, -181,
+  315, 184, 53, -57, -188,
+  329, 191, 53, -57, -195,
+  343, 198, 53, -57, -202,
+  347, 200, 53, -57, -204
+};
+std::array<float, 6> LTUParam::mgX = { 300.65, 313.25, 325.85, 338.45, 351.05, 363.65 };
+std::array<float, 6> LTUParam::mgTiltingAngle = { -2., 2., -2., 2., -2., 2. };
+int LTUParam::mgDyMax = 63;
+int LTUParam::mgDyMin = -64;
+float LTUParam::mgBinDy = 140e-4;
+std::array<float, 6> LTUParam::mgWidthPad = { 0.635, 0.665, 0.695, 0.725, 0.755, 0.785 };
+std::array<float, 6> LTUParam::mgLengthInnerPadC1 = { 7.5, 7.5, 8.0, 8.5, 9.0, 9.0 };
+std::array<float, 6> LTUParam::mgLengthOuterPadC1 = { 7.5, 7.5, 7.5, 7.5, 7.5, 8.5 };
+std::array<float, 6> LTUParam::mgInvX;
+std::array<float, 6> LTUParam::mgTiltingAngleTan;
+std::array<float, 6> LTUParam::mgInvWidthPad;
+
+float LTUParam::mgLengthInnerPadC0 = 9.0;
+float LTUParam::mgLengthOuterPadC0 = 8.0;
+float LTUParam::mgScalePad = 256. * 32.;
+float LTUParam::mgDriftLength = 3.;
+
+LTUParam::LTUParam() : mMagField(0.),
+                       mOmegaTau(0.),
+                       mPtMin(0.1),
+                       mNtimebins(20 << 5),
+                       mScaleQ0(0),
+                       mScaleQ1(0),
+                       mPidTracklengthCorr(false),
+                       mTiltCorr(false),
+                       mPidGainCorr(false)
+{
+  // default constructor
+  // These variables are used internally in the class to elliminate divisions.
+  // putting them at the top was messy.
+  int j = 0;
+  std::for_each(mgInvX.begin(), mgInvX.end(), [&j](float& x) { 1 / mgX[j]; });
```

Comment:
should this be `x = 1./mgX[j]` ? (there is a compiler warning here)

Reply:
yes, I ended up just changing it to a single for loop and doing all 3 at each step.